elan/rpc/rclient.go

```diff
@@ -33,39 +33,53 @@ func (r *remoteClient) Healthcheck() error {
 }
 
 func (r *remoteClient) ReadBlob(dg *pb.Digest) ([]byte, error) {
-	return r.c.ReadBlob(context.Background(), digest.NewFromProtoUnvalidated(dg))
+	ctx, cnx := context.WithTimeout(context.Background(), time.Minute*2)
+	defer cnx()
+	return r.c.ReadBlob(ctx, digest.NewFromProtoUnvalidated(dg))
 }
 
 func (r *remoteClient) WriteBlob(b []byte) (*pb.Digest, error) {
-	dg, err := r.c.WriteBlob(context.Background(), b)
+	ctx, cnx := context.WithTimeout(context.Background(), time.Minute*2)
+	defer cnx()
+	dg, err := r.c.WriteBlob(ctx, b)
 	return dg.ToProto(), err
 }
 
 func (r *remoteClient) UpdateActionResult(req *pb.UpdateActionResultRequest) (*pb.ActionResult, error) {
-	return r.c.UpdateActionResult(context.Background(), req)
+	ctx, cnx := context.WithTimeout(context.Background(), time.Minute*2)
+	defer cnx()
+	return r.c.UpdateActionResult(ctx, req)
 }
 
 func (r *remoteClient) UploadIfMissing(entries []*uploadinfo.Entry) error {
-	_, _, err := r.c.UploadIfMissing(context.Background(), entries...)
+	ctx, cnx := context.WithTimeout(context.Background(), time.Minute*2)
+	defer cnx()
+	_, _, err := r.c.UploadIfMissing(ctx, entries...)
```

Comment:
I would be a bit worried about this one. `UploadIfMissing` can potentially make a bunch of internal RPCs and while I'd generally expect us to upload outputs in under 2 minutes total, I can imagine some things (aws-sdk-go...) being slow.

Reply:
Cool, will increase the two timeouts you had concerns about, and also whack in the metrics so we can visibly see when it's timing out.